net-dns/yaddns/patches/yaddns-1.1.patchset

```diff
@@ -0,0 +1,238 @@
+From 13408140003c2f0be1d8fd03d12a6751bd07e75d Mon Sep 17 00:00:00 2001
+From: sfanxiang <sfanxiang@gmail.com>
+Date: Mon, 8 Jan 2018 10:10:48 +0000
+Subject: port to Haiku
+
+
+diff --git a/configure.in b/configure.in
+index d8c6668..fd6b42d 100644
+--- a/configure.in
++++ b/configure.in
+@@ -9,6 +9,7 @@ test ".$CFLAGS" = "." && CFLAGS=" "
+ 
+ # programs needed to build/install
+ AC_PROG_CC
++AC_PROG_CXX
+ AC_PROG_INSTALL
+ AC_PROG_RANLIB
+ 
+@@ -66,6 +67,8 @@ if test "x$enable_log_color" = "xyes"; then
+    CFLAGS="$CFLAGS -DENABLE_LOG_COLOR"
+ fi
+ 
++CXXFLAGS=$CFLAGS
++
+ # the generated files
+ AC_CONFIG_FILES([
+ Makefile
+diff --git a/src/Makefile.am b/src/Makefile.am
+index 750ae3a..122a989 100644
+--- a/src/Makefile.am
++++ b/src/Makefile.am
+@@ -5,6 +5,7 @@ bin_PROGRAMS = yaddns
+ yaddns_SOURCES = yaddns.c yaddns.h \
+ 	config.c config.h \
+ 	account.c account.h \
++	ifaddr.cc ifaddr.h \
+ 	request.c request.h \
+ 	log.c log.h \
+ 	util.c util.h \
+diff --git a/src/config.c b/src/config.c
+index 8f7b324..99eab8c 100644
+--- a/src/config.c
++++ b/src/config.c
+@@ -7,13 +7,15 @@
+ #include <getopt.h>
+ #include <limits.h>
+ 
++#include <FindDirectory.h>
++#include <fs_info.h>
++
+ #include "config.h"
+ #include "log.h"
+ #include "service.h"
+ #include "services.h"
+ #include "util.h"
+ 
+-#define CFG_DEFAULT_FILENAME "/etc/yaddns.conf"
+ #define CFG_DEFAULT_WANIFNAME "ppp0"
+ 
+ /*
+@@ -228,7 +230,18 @@ int config_parse(struct cfg *cfg, int argc, char **argv)
+ 
+         if(!cfgstr_is_set(&(cfg->cfgfile)))
+         {
+-                cfgstr_set(&(cfg->cfgfile), CFG_DEFAULT_FILENAME);
++                const char filename[] = "/yaddns/yaddns.conf";
```

Comment:
```diff
-                const char filename[] = "/yaddns/yaddns.conf"
+                const char filename[] = "/yaddns.conf"
```

Reply:
Don't forget to replace `"/yaddns/yaddns.conf"` by `"/yaddns.conf"`, otherwise yaddns will look for `/boot/system/settings/yaddns/yaddns.conf` instead of `/boot/system/settings/yaddns.conf`.
For the `TEST()` function, what @alaviss [suggests](https://github.com/haikuports/haikuports/pull/2086#discussion_r160634952) is that you add this after `INSTALL()`:
```
TEST()
{
	make check
}
```